### ISSUE TITLE
Pointer to defined variables

### ISSUE THREAD

**Issue by jimmyhedstr0m** [Koromix/koffi#80]
I'm trying to run a Simulink model exported as a DLL file together with koffi. However, from the documentation and severals different attempts, I don't understand how to get pointers to variables declared in the root file. All examples I have seen have involved pointers in various parameters which you can set and get values from. What I aim for is to be able to set and get values from structs directly in the root file in order to feed input to the Simulink model and then read the outputs.

Here is are the relevant parts of the corresponding C code to make things more clear:
```
// model.h

typedef double real_T;

typedef struct {
  real_T In1;
} ExtU_model_T;

typedef struct {
  real_T Out1;
  real_T Out2;
} ExtY_model_T;

extern ExtU_model_T model_U;

extern ExtY_model_T model_Y;

extern void model_initialize(void);
extern void model_step(void);
extern void model_terminate(void);


// model.c

// Model input
ExtU_model_T model_U;

// Model output
ExtY_model_T model_Y;

void model_step(void)
{
  model_Y.Out2 = 2.0; // simplified example with hardcoded double here
  model_Y.Out1 = model_Y.Out2 * model_U.In1;

  // ...
}

void model_initialize(void)
{
  // ...
}

void model_terminate(void)
{
  // ...
}
```

And here are my attempts with some comments of what I would like to achieve:
```
// index.ts
import koffi from 'koffi';

const lib = koffi.load('./model_win64.dll');

const initialize = lib.func('model_initialize', 'void', []);
const step       = lib.func('model_step', 'void', []);
const terminate  = lib.func('model_terminate', 'void', []);

koffi.alias('real_T', 'double');

const ExtU_model_T = koffi.struct('ExtU_model_T', {
  In1: 'real_T',
});

const ExtY_model_T = koffi.struct('ExtY_model_T', {
  Out1: 'real_T',
  Out2: 'real_T',
});

const inputPointer  = koffi.pointer('model_U', ExtU_model_T, 1);
const outputPointer = koffi.pointer('model_Y', ExtY_model_T, 1);

// How to I assign a value to model input like in C, model_U.In1 = 1337.0;?

model_initialize();
model_step();

// Trying to read output values
const output = koffi.decode(outputPointer, ExtY_model_T);
console.log(output);
// Prints something like this, which isn't correct
// {
//   Out1: 1.1113799477324e-311,
//   Out2: 1.69759663366e-313
// }

model_terminate();
```

I would really appreciate further guidance here please 🙏

**Comment by neko-para** [comment]
Actually, the inputPointer and outputPointer you've created are types but not variables.
```cpp
typedef ExtU_model_T* model_U; // const inputPointer  = koffi.pointer('model_U', ExtU_model_T, 1);
```

---

Currently, seems that koffi doesn't support getting a variable symbol from the library. You may edit the library and export the variable as the return value of a function.

Meanwhile, according to #77, there would be a `koffi.encode` in the future, which helps you write data into a pointer. But as this function hasn't been done, you could use `memcpy` instead.

**Comment by jimmyhedstr0m** [comment]
Thanks for the clarification! My guess was that it was the pointer address I got instead of the value due to exponent -313.

Edit the library is not an option since the DLL file is auto-generated from Simulink upon model export. The only option as I see it is to write a wrapper around the DLL library in C, which will be loaded as a native Node addon, but that kinda defeats the purpose of this library unfortunately.

**Comment by neko-para** [comment]
After all, you can simply write a C library that export a function return the address of model_*.

**Comment by Koromix** [comment]
I have added `koffi.symbol()` in Koffi 2.5.21-beta.3, along with `koffi.encode()`.
Install with `npm install koffi@beta`.

It is not documented yet, but you can find a small example extracted from the test suite below.

```js
// let lib = koffi.load(lib_filename);

const sym_int = lib.symbol('sym_int', 'int');
const sym_str = lib.symbol('sym_str', 'const char *');
const GetSymbolInt = lib.func('int GetSymbolInt()');
const GetSymbolStr = lib.func('const char *GetSymbolStr()');

koffi.encode(sym_int, 'int', 12);
koffi.encode(sym_str, 'const char *', 'I think...');
koffi.encode(sym_str, 'const char *', 'I can encode!');
assert.equal(GetSymbolInt(), 12);
assert.equal(GetSymbolStr(), 'I can encode!');
```

```c
int sym_int = 0;
const char *sym_str = NULL;

int GetSymbolInt()
{
    return sym_int;
}

const char *GetSymbolStr()
{
    return sym_str;
}
```

**Comment by Koromix** [comment]
I've published Koffi 2.6.0 with all of this.

Documentation: https://koffi.dev/variables
Changelog: https://koffi.dev/changelog
